### PR TITLE
Guest dropdown

### PR DESCRIPTION
**Pull request by cshih1141**
added functionality and styling for guests dropdown. database hasn't been linked yet.

### CODE REVIEW

File: public/style.css [hunk 1]
```diff
@@ -234,3 +236,195 @@
...
+#adultAdd, #adultMinus, #adultCounter {
+  grid-row-start: 1;
+  grid-row-end: 2;
+}
+
+#adultGuest {
+  margin-top: 20px;
+}
+
+/* Popup arrow */
```
Comment (galw) [inline public/style.css]: Should be deleted before pull request

File: public/style.css [hunk 2]
```diff
@@ -234,3 +236,195 @@
...
+  align-self: center;
+}
+
+.typeOfGuest {
+  display: grid;
+  grid-template-rows: 25px 45px; 
+  grid-template-columns: 165px 40px 45px 40px;
+}
+
+#adultGuestDropdown {
```
Comment (galw) [inline public/style.css]: As a better practice, when writing css try to use mainly classes and html tags instead if id's, for example what if later your will find our that you will have another drop down that should have exactly the same style? or if you'll have two of there.

File: public/style.css [hunk 3]
```diff
@@ -234,3 +236,195 @@
...
+  grid-template-rows: 70px; 
+}
+
+#adultAdd, #adultMinus, #adultCounter {
```
Comment (galw) [inline public/style.css]: Why not to put them in css class together?

File: public/style.css (hunk 1 above)
Comment (cshih1141) [reply public/style.css]: i wanted to keep that there for reference as I may need it later

File: public/style.css (hunk 2 above)
Comment (cshih1141) [reply public/style.css]: I'll try, but I think like 90% of my css are classes (.) is for class i think

File: client/src/App.jsx [hunk 4]
```diff
@@ -4,23 +4,75 @@ import Calendar from './Calendar';
...
+  componentDidMount() {
+    document.addEventListener('click', this.handleClick, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClick, false);
   }
 
   render() {
     return(
```
Comment (LittleJennie) [inline client/src/App.jsx]: Feel free to refactor based on the comments in later commits.

You can destructure `this.state`'s keys within render before you return.

File: client/src/App.jsx [hunk 5]
```diff
@@ -4,23 +4,75 @@ import Calendar from './Calendar';
 import Guests from './Guests';
 import Reserve from './Reserve';
 
-
 class App extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      guestDropdownVisibility: "hidden",
```
Comment (LittleJennie) [inline client/src/App.jsx]: Looks like you are passing many of these states more than 1 level down. Is there ways we can refactor to have fewer of them in App.jsx?

File: client/src/App.jsx [hunk 6]
```diff
@@ -31,6 +83,161 @@ class App extends React.Component {
...
+        let numAdults = this.state.numAdults + 1;
+        this.setState({
+          numAdults
+        }, this.updateButtonStatus);
+      } else if (type === "children") {
+        let numChildren = this.state.numChildren + 1;
+        this.setState({
+          numChildren
+        }, this.updateButtonStatus);
+      }
```
Comment (LittleJennie) [inline client/src/App.jsx]: Repetitive codes. Please combined, maybe create a function for it

File: client/src/App.jsx [hunk 7]
```diff
@@ -31,6 +83,161 @@ class App extends React.Component {
...
+        let numChildren = this.state.numChildren - 1;
+        this.setState({
+          numChildren
+        }, this.updateButtonStatus);
+      } else if(type === "infant" && this.state.numInfants > 0) {
+        let numInfants = this.state.numInfants - 1;
+        this.setState({
+          numInfants
+        }, this.updateButtonStatus);
+      }
```
Comment (LittleJennie) [inline client/src/App.jsx]: same as above, repetitive, please refactor

File: client/src/App.jsx [hunk 8]
```diff
@@ -4,23 +4,75 @@ import Calendar from './Calendar';
...
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleClick, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClick, false);
   }
 
   render() {
```
Comment (LittleJennie) [inline client/src/App.jsx]: `render` should be the last function in the class

File: client/src/GuestsDropdown.jsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+      <span className="guestCounter" id="childCounter">{numChildren}</span>
+      <button className="addButton" id="childAdd" disabled={disableChildrenPlus} onClick={() => incrementGuestsCounter("children")}>+</button>
+    </div>
+    <div className="typeOfGuest" id="infantGuestDropdown">
+      <span className="guestType" id="infantGuest">Infants</span>
+      <span className="guestDetails" id="infantDetails">Under 2</span>
+      <button className="minusButton" id="infantMinus" disabled={disableInfantMinus} onClick={() => decrementGuestsCounter("infant")}>-</button>
+      <span className="guestCounter" id="infantCounter">{numInfants}</span>
+      <button className="addButton" id="infantAdd" disabled={disableInfantPlus} onClick={() => incrementGuestsCounter("infant")}>+</button>
+    </div>
```
Comment (LittleJennie) [inline client/src/GuestsDropdown.jsx]: Repetitive codes, please create a separate component to reuse for rendering.

**Pull request merged**